Sources/Mapper.swift

```diff
@@ -314,7 +314,7 @@ public struct Mapper {
 
         var result = [U: T]()
         for (key, value) in data {
-            result[try U.fromMap(key)] = try T.fromMap(value)
+            result[try U.fromMap(key as AnyObject)] = try T.fromMap(value as AnyObject)
```

Comment:
Yeah perhaps we should. The intention of this PR was really just to make this branch build again and not to rethink APIs for Swift 3 (though I definitely think we should do that too).